### ISSUE TITLE
Penalty system

### ISSUE THREAD

**Issue by Kayowin** [MarcTowler/discord-bot#81]
a multi tier, multi layer penalty system unique to user with the ability to issue, revoke and query instances.

**Layers:**
**Penalty by infraction class:**

- Event non attendance (everyone role upward inc council)
- Conduct (everyone role upward inc council)
- Event Requirement (officer role only)

**Tiers
Level**

**Event non attendance** 
1 - penalty point logged against user ID - PM sent with reason
2 - penalty point logged against user ID - PM sent with reason
3 - penalty point logged against user ID - PM sent with reason, roles removed & sent to phantom zone.

**Conduct** 
1 - initial warning logged against user ID - PM sent with reason
2 - final warning logged against user ID - PM sent with reason
3 - removal notice logged against user ID - PM sent with reason, user roles removed & banned.

**Event Requirement**
1 - penalty point logged against user ID - PM sent with reason
2 - penalty point logged against user ID - PM sent with reason
3 - demotion notice logged against user ID - PM sent with reason, officer role removed.

**Infraction query/lookup**
- By user ID
- All
- by role (a nice to have but not essential)

**Infraction revoke**
- By user instance <tier><layer>
- All

**Comment by MarcTowler** [comment]
![image](https://user-images.githubusercontent.com/1121980/58338258-57a54880-7e3f-11e9-85c3-d1a4983d192d.png)
this is a layout idea that was shared in discord

**Comment by MarcTowler** [comment]
Completed on stream this evening, full info as shown in the example image was not required at this moment... Waiting on Council to push live
